sidekick_core/lib/src/commands/format_command.dart

```diff
@@ -0,0 +1,203 @@
+    if (packageName != null) {
+      final package =
+          allPackages.where((it) => it.name == packageName).firstOrNull;
+      if (package == null) {
+        throw "Package with name $packageName not found in repository "
+            "${SidekickContext.repository?.path}";
+      }
+      // only format for selected package
+      // _format(package, globalLineLength: lineLength);
+      return;
+    }
+    final globExcludes = excludeGlob
+        .expand((rule) {
+          // start search at repo root
+          final root = SidekickContext.repository?.path;
+          return Glob("$root/$rule").listSync(root: root);
+        })
+        .whereType<Directory>()
+        .mapNotNull((e) => DartPackage.fromDirectory(e));
+
+    final excluded = [
+      ...exclude,
+      ...globExcludes,
+    ];
+
+    // Key: line length
+    // Value: all files to be formatted with the line length specified by key
+
+    final lineLengthsAndFiles = <int, List<File>>{};
+    for (final package
+        in allPackages.filter((package) => !excluded.contains(package))) {
+      final lineLength = getLineLength(package);
+      final allFilesInPackage = package.root
+          .listSync(recursive: true)
+          .whereType<File>()
+          .filter((file) => file.extension == '.dart')
+          .filter(
+            (file) => file.uri.pathSegments.none(
+              (element) => element.startsWith('.'),
+            ),
+          )
+          .filter((file) {
+        // exclude files from packages nested inside the current package
+        //
+        // e.g.
+        // package_bar: 80
+        //   package_bar_example: 120
+        //
+        // if this step was omitted, the result would be
+        // {
+        //   80: [package_bar/main.dart, package_bar/example/main.dart, ...],
+        //   120: [package_bar/example/main.dart, ...],
+        // }
+        // that is wrong, the correct result is
+        // {
+        //   80: [package_bar/main.dart, ...],
+        //   120: [package_bar/example/main.dart, ...],
+        // }
+
+        // get all packages except the current package in iteration
+        final allOtherPackages = ([...allPackages]..remove(package));
+        return allOtherPackages.any((otherPackage) {
+          // does any of the other packages also contain the current file?
+          if (file.path.contains(otherPackage.root.path)) {
+            // exclude file if path of other package matches the file path more closely
+            //
+            // e.g.
+            // package: packages/bar
+            // otherPackage: packages/bar/example
+            // file: packages/bar/example/main.dart
+            //
+            // otherPackage matches file path more closely,
+            // so the file should be excluded for the current package
+
+            return otherPackage.root.path.length < package.root.path.length;
+          }
+          return true;
+        });
```

Comment:
This seems unnecessarily expensive. For every package, every file, will be checked against every package. There are many unnecessary checks and it gets worse for every nesting layer.

Better way: 
- find all files [all]
- Sort the package paths by length
- Start with the longest path
  - Remove files from [all] that belong in path
  - Save in [map] with package -> files
- Iterate over packages in [map] and do work for each file